OpenRA.Mods.Common/Irc.cs

```diff
@@ -0,0 +1,21 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Threading;
+using Meebey.SmartIrc4net;
+
+namespace OpenRA.Mods.Common
+{
+	public static class Irc
+	{
+		public static IrcClient Client;
```

Comment:
That was just an example, any operation where we check multiple things on the UI thread if they could change in the meantime might go wrong. Another example:
```
if (Irc.Client.JoinedChannels.Count > 0)
    channel = Irc.Client.GetChannel(Irc.Client.JoinedChannels[0]);
```

Let's say the IRC thread leaves the channel after our check but before we get the channel. We'd hit an exception at this point. Something needs to be done to ensure race conditions like this don't bite us.